src/Microsoft.OData.Core/ODataMetadataJsonInputContext.cs

```diff
@@ -0,0 +1,99 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ODataMetadataJsonInputContext.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+#if NETSTANDARD2_0
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+using Microsoft.OData.Edm;
+using Microsoft.OData.Edm.Csdl;
+using Microsoft.OData.Edm.Validation;
+
+namespace Microsoft.OData
+{
+    /// <summary>
+    /// Implementation of the OData input for JSON metadata documents.
+    /// </summary>
+    internal sealed class ODataMetadataJsonInputContext : ODataInputContext
+    {
+        /// <summary>The stream to read from.</summary>
+        private Stream messageStream;
+
+        /// <summary>Constructor.</summary>
+        /// <param name="messageInfo">The context information for the message.</param>
+        /// <param name="messageReaderSettings">Configuration settings of the OData reader.</param>
+        public ODataMetadataJsonInputContext(
+            ODataMessageInfo messageInfo,
+            ODataMessageReaderSettings messageReaderSettings)
+            : base(ODataFormat.Metadata, messageInfo, messageReaderSettings)
+        {
+            Debug.Assert(messageInfo.MessageStream != null, "messageInfo.MessageStream != null");
+            messageStream = messageInfo.MessageStream;
+        }
+
+        /// <summary>
+        /// Read a metadata document.
+        /// This method reads the metadata document from the input and returns
+        /// an <see cref="IEdmModel"/> that represents the read metadata document.
+        /// </summary>
+        /// <returns>An <see cref="IEdmModel"/> representing the read metadata document.</returns>
+        internal override IEdmModel ReadMetadataDocument()
+        {
+            // We can't use stream.Read(Span<byte> buffer), this method is introduced since .NET Core 2.1. :(
+            byte[] bytes = this.messageStream.ReadAllBytes();

```

Comment:
Is it worth having an #IFDEF for later versions of .NET Core?

Reply:
At top this class, we have "#if NETSTANDARD2_0".

Besides, we don't output for other version of .NET Core. So, can we leave it as it?

---
In reply to: [518471948](https://github.com/OData/odata.net/pull/1913#discussion_r518471948) [](ancestors = 518471948)